lib/screens/configure_authentication/screen.dart

```diff
@@ -0,0 +1,197 @@
+import 'package:concordium_wallet/widgets/base_page.dart';
+import 'package:concordium_wallet/state/auth.dart';
+import 'package:concordium_wallet/state/services.dart';
+import 'package:concordium_wallet/widgets/toggle.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:flutter_svg/flutter_svg.dart';
+
+class AuthenticationData {
+  final String password;
+  final bool enableBiometrics;
+
+  const AuthenticationData({required this.password, required this.enableBiometrics});
+}
+
+class ConfigureAuthenticationScreen extends StatelessWidget {
+  const ConfigureAuthenticationScreen({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    return BasePage(
+      title: 'Security',
+      body: ConfigureAuthenticationForm(
+        onContinuePressed: (data) async {
+          final services = context.read<ServiceRepository>();
+          final authentication = context.read<Authentication>();
+
+          // Persist password and then register the user as authenticated.
+          await services.auth.setPassword(data.password);
+          authentication.setAuthenticated(true);
+
+          // TODO: Enable biometrics if requested.
+
+          // Navigate back to home screen (mount check necessary because of the "await barrier").
+          if (!context.mounted) return;
+          Navigator.pop(context);
+        },
+      ),
+    );
+  }
+}
+
+class ConfigureAuthenticationForm extends StatefulWidget {
+  final void Function(AuthenticationData data) onContinuePressed;
+
+  const ConfigureAuthenticationForm({super.key, required this.onContinuePressed});
+
+  @override
+  State<ConfigureAuthenticationForm> createState() => _ConfigureAuthenticationFormState();
+}
+
+class _ConfigureAuthenticationFormState extends State<ConfigureAuthenticationForm> {
+  final _passwordController1 = TextEditingController();
+  final _passwordController2 = TextEditingController();
+
+  var _password1 = '';
+  var _password2 = '';
```

Comment:
Why can’t we just use the text field from the controllers?

Reply:
The widget doesn't automatically update (i.e. change disabled state of the continue button) when the controller input changes. We need to update some widget state to make it re-render. At least I'm not aware of other ways to make it do that.